machine/clients/slack.py

```diff
@@ -93,6 +116,71 @@ async def _process_users_channels(self, _: AsyncBaseSocketModeClient, req: Socke
             elif event["type"] == "member_joined_channel":
                 await self._on_member_joined_channel(event)
 
+    @rate_limit_retry
```

Comment:
Dumb question: why did you put the retry logic in a decorator? To me, that makes the code harder to read, because the logic for calling the API now lives in 2 places. The decorator is not reused because `_fetch_paginated_data` is already a generic reusable function in itself.

Reply:
Github is shit and isn't updating the diff. I was messing around with the idea, but quickly reverted it after coming to the same conclusion.

![Screenshot 2024-11-14 at 2 32 03 PM](https://github.com/user-attachments/assets/d8db405d-c2f6-42c8-b2f7-5aa2752d6691)

Reply:
I'm going to re-open a new PR.